### PR TITLE
Changelog update + CI Fix

### PR DESCRIPTION
**Pull request by doctrino**
## Description
Changelog updates and special treatment of Groups to run first to give general access, and then the scoped access after all resources have been created. 

Tested locally with a Service Principal with only Projects and Groups capabilities. 

## Checklist:
- [ ] Tests added/updated.
- [ ] Documentation updated.
- [x] Changelogs updated in [CHANGELOG.cdf-tk.md](https://github.com/cognitedata/cdf-project-templates/blob/main/CHANGELOG.cdf-tk.md).
- [x] Template changelogs updated in [CHANGELOG.templates.md](https://github.com/cognitedata/cdf-project-templates/blob/main/CHANGELOG.templates.md).
- [ ] Version bumped. [version.py](https://github.com/cognitedata/cdf-project-templates/blob/main/cognite/cognite_toolkit/_version.py) and
  [pyproject.toml](https://github.com/cognitedata/cdf-project-templates/blob/main/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: cognite_toolkit/cdf.py [hunk 1]
```diff
@@ -261,6 +262,20 @@ def deploy(
         )
         exit(1)
     print(ToolGlobals.as_string())
+    if "auth" in include and (directory := (Path(build_dir) / "auth")).is_dir():
+        # First, we need to get all the generic access, so we can create the rest of the resources.
+        drop_load_resources(
+            AuthLoader.create_loader(ToolGlobals, load="all_scoped_only"),
```
Comment (gregertw) [inline cognite_toolkit/cdf.py:268]: Neat

File: cognite_toolkit/cdf.py [hunk 2]
```diff
@@ -431,9 +459,9 @@ def clean(
             if ToolGlobals.failed:
                 print(f"[bold red]ERROR: [/] Failure to clean {LoaderCls.folder_name} as expected.")
                 exit(1)
-    if "auth" in include:
+    if "auth" in include and (directory := (Path(build_dir) / "auth")).is_dir():
         drop_load_resources(
-            LOADER_BY_FOLDER_NAME.get("auth"),
+            AuthLoader.create_loader(ToolGlobals, load="all"),
```
Comment (gregertw) [inline cognite_toolkit/cdf.py:464]: Here the "load" var seems a bit wrong, maybe target would be a more appropriate name?

File: cognite_toolkit/cdf_tk/utils.py [hunk 3]
```diff
@@ -281,7 +281,7 @@ def verify_client(
             raise ValueError(f"Failed to load capabilities from {capabilities}. Wrong syntax?")
         comp = self.client.iam.compare_capabilities(resp.capabilities, caps)
         if len(comp) > 0:
-            print(f"Capabilities mismatch: {comp}")
+            print(f"Missing Capabilities: {comp}")
```
Comment (gregertw) [inline cognite_toolkit/cdf_tk/utils.py:284]:
```suggestion
            print(f"Missing necessary CDF access capabilities: {comp}")
```

**Review by gregertw — APPROVED** [review]
Super! Added some comments, approving so you can merge when ready.

File: cognite_toolkit/cdf.py (hunk 2 above)
Comment (doctrino) [reply cognite_toolkit/cdf.py]: Renamed it to `target_scopes`

**Pull request merged**